cli/flox-activations/src/cli/activate/mod.rs

```diff
@@ -70,6 +80,15 @@ impl ActivateArgs {
             subsystem_verbosity,
             VarsFromEnvironment::get()?,
         );
+        let env_diff = EnvDiff::from_files(&activation_state_dir)?;
+        let startup_ctx = Self::startup_ctx(
+            context.clone(),
+            invocation_type,
+            env_diff,
+            &activation_state_dir,
+        )?;
+        // Writes to either a startup file or stdout, depending on invocation type
+        Self::write_rc_script(&startup_ctx)?;
```

Comment:
style nonblocking: I think it's a bit opaque to have startup_ctx() controlling where the startup script gets written. If I go read activate_in_place() it's pretty unclear where the script came from. I think it would be more clear to have:
```
let startup_ctx = startup_ctx();
activate_in_place(startup_ctx);
fn activate_in_place() {
  script = generate_startup_commands(startup_ctx, cleanup = None)
  writeln(script)
}
```

Reply:
I originally had `generate_*` doing that, but that's worse for testing because then `generate_*` actually writes to a file rather than just writing a string to a generic writer (which can be a `String`/`Vec<u8>` in testing).

I strongly prefer that the `generate_*` functions write to a generic writer, but it doesn't have to be `StartupCtx` that determines the path. That could be pulled out into `handle`. I do think it's best that `StartupCtx` contain whatever information is necessary in order to produce the script though (regardless of where it gets written). I'm trying to keep it such that `generate_*` functions are as agnostic as possible to how they're being used so that we can consolidate their logic over time.

Reply:
I was thinking `generate_*` could just return a string (hence the writeln). The POC has :
```
        let script = formatdoc! {r#"
            {legacy_exports}
            {flox_activations} attach --runtime-dir "{runtime_dir}" --pid {self_pid_var} --flox-env "{flox_env}" --id "{id}" --remove-pid "{pid}";
            {startup_commands}
        "#, ...
        };
        print!("{script}");
```

If generate_* actually writes to stdout, we'll have to call it after we print legacy exports. Which means we'll have to call generate_* in each different activate function for each invocation type

Reply:
I'm not really sure what those legacy exports are, but I don't think this is the worst thing (keep most of the logic invocation-agnostic, branch out where it needs to be different):

```
let startup_ctx = startup_ctx();

match act_ctx.invocation_type {
    InPlace => { do_exports() },
    Command => { do_exports() },
    Interactive => { do_exports() },
}

write_rc_script();

match act_ctx.invocation_type {
    InPlace => { run_cmd() },
    Command => { run_cmd() },
    Interactive => { run_cmd() },
}
```